src/Chapter2/genFibo2.hs

```diff
@@ -0,0 +1,24 @@
+module Chapter2.GenFibo2 where
+import Data.Char
+import Data.List
+
+fibo :: Int -> Int
+fibo 0 = 0
+fibo 1 = 1
+fibo n = fibo (n-1) + fibo (n-2)
+narr :: Int -> Int -> [Int]
+narr num n
+    |num < n = [fibo num] ++ (narr (num+1) n)
+    |num == n = [fibo num]
+
+genFibo :: (Int -> Bool) -> Int -> [Int]
+genFibo fn n =  (take n (filter (\num -> fn num) (narr 1 (n*10))))
```

Comment:
why are you doing `n*10`? we discussed this in the morning -- you have to implement this using the unified fibo function that you solved in the previous chapter!